sled-agent/src/services.rs

```diff
@@ -89,6 +94,13 @@ pub enum Error {
 
     #[error("Services already configured for this Sled Agent")]
     ServicesAlreadyConfigured,
+
+    #[error("Error resolving DNS name: {0}")]
+    ResolveError(#[from] ResolveError),
+
+    // TODO: Remove this error; prefer to retry notifications.
+    #[error("Notifying Nexus failed: {0}")]
+    Notification(#[from] nexus_client::Error<nexus_client::types::Error>),
```

Comment:
Can we remove this before merging? Doesn't the retry request try forever?

Reply:
It does - I created a separate Error type in b2df602c, along with a note that both errors are transient and should be retried indefinitely.